src/demo/renderer.js

```diff
@@ -56,27 +56,32 @@ function loadBackgroundImage() {
   });
 }
 
-function renderBackgroundImage(ctx, img) {
+function renderBackgroundImage(img) {
+  const backgroundCanvas = document.getElementById('background-canvas');
+  if (!backgroundCanvas) {
+    return;
+  }
+
+  backgroundCanvas.width = constants.canvasWidth;
+  backgroundCanvas.height = constants.canvasHeight;
+  const ctx = backgroundCanvas.getContext('2d');
   ctx.drawImage(img, 0, 0, constants.canvasWidth, constants.canvasHeight);
 }
 
+function initTrainingScreen(state) {
+  // We only want to draw the background and UI elements once,
+  // so render here instead of in drawTrainingScreen().
+  loadBackgroundImage().then(backgroundImg => {
+    renderBackgroundImage(backgroundImg);
+    drawTrainingScreen(state);
+    drawTrainingUiElements(state);
```

Comment:
These are more examples of where we should just set some state back in our main app, and then let the renderer respond to that state change when it next detects that it's happened.